test/test_ops.py

```diff
@@ -1446,6 +1447,18 @@ def _tensor_requires_grad(x):
 
         self.fail(msg)
 
+    # Validates that each OpInfo that sets promotes_int_to_float=True does as it says
+    @skipMeta
+    @onlyNativeDeviceTypes
+    @ops((op for op in op_db if op.promotes_int_to_float), allowed_dtypes=integral_types_and(torch.bool))
+    def test_promotes_int_to_float(self, device, dtype, op):
+        for sample in op.sample_inputs(device, dtype):
+            output = op(sample.input, *sample.args, **sample.kwargs)
+            if not output.dtype.is_floating_point:
+                self.fail(
```

Comment:
self.assertTrue(output.dtype.is_floating_point

Reply:
This creates the ugly prefix: `AssertionError: False is not True`. And and the 

```python
if not ...:
    self.fail()
```

doesn't really add any complexity.

Reply:
We should then overload assertTrue as `if not ...` for cleaner errors all throughout. Not necessarily in this PR though.

In general, it's just nicer if there's just one right way of doing things.

Reply:
> In general, it's just nicer if there's just one right way of doing things.

Just a note from a "veteran" in trying to refactor the test suite: if you want to preserve your sanity, don't even try. You'll waste tens of hours to establish a standard that no one will follow anyway.